plugins/woocommerce/client/blocks/assets/js/blocks/product-details/test/block.tsx

```diff
@@ -0,0 +1,202 @@
+/**
+ * External dependencies
+ */
+import '@testing-library/jest-dom';
+import { screen, waitFor, within } from '@testing-library/react';
+import { createBlock, type BlockAttributes } from '@wordpress/blocks';
+import { http, HttpResponse } from 'msw';
+import { setupServer } from 'msw/node';
+import { useSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import '../';
+import '../../accordion/accordion-group';
+import '../../accordion/inner-blocks/accordion-header';
+import '../../accordion/inner-blocks/accordion-item';
+import '../../accordion/inner-blocks/accordion-panel';
+import '../../product-description';
+import '../../product-reviews';
+import '../../product-specifications';
+import '../../single-product';
+
+import { initializeEditor } from '../../../../../tests/integration/helpers/integration-test-editor';
+import {
+	productWithoutSpecifications,
+	productWithSpecifications,
+} from '../fixture';
+
+async function setupWithSingleProduct(
+	attributes: BlockAttributes,
+	productId: number
+) {
+	const productDetailsBlock = createBlock(
+		'woocommerce/product-details',
+		attributes
+	);
+
+	const singleProductBlock = [
+		{
+			name: 'woocommerce/single-product',
+			attributes: {
+				productId,
+			},
+			innerBlocks: [ productDetailsBlock ],
+		},
+	];
+
+	return initializeEditor( singleProductBlock );
+}
+
+jest.mock( '@wordpress/data', () => ( {
+	...jest.requireActual( '@wordpress/data' ),
+	useSelect: jest.fn(),
+} ) );
+
+describe( 'Product Details block', () => {
+	describe( 'Single Product block', () => {
+		const server = setupServer(
+			http.get( '/wc-admin/options', ( { request } ) => {
+				const url = new URL( request.url );
+				const options = url.searchParams.get( 'options' );
+				// Check if the request is for dimension and weight units
+				if (
+					options ===
+					'woocommerce_dimension_unit,woocommerce_weight_unit'
+				) {
+					return HttpResponse.json( {
+						woocommerce_dimension_unit: 'cm',
+						woocommerce_weight_unit: 'kg',
+					} );
+				}
+				// Default response for other options requests
+				return HttpResponse.json( {}, { status: 200 } );
+			} ),
+			http.get( '/wc/store/v1/products/*', () => {
+				return HttpResponse.json( productWithSpecifications );
+			} ),
```

Comment:
Nit: I'm compelled to make some of these one-liners:

```suggestion
			http.get( '/wc/store/v1/products/*', () => HttpResponse.json( productWithSpecifications ) ),
```